module/os/linux/zfs/zpl_super.c

```diff
@@ -117,8 +154,25 @@ zpl_sync_fs(struct super_block *sb, int wait)
 	error = -zfs_sync(sb, wait, cr);
 	spl_fstrans_unmark(cookie);
 	crfree(cr);
```

Comment:
I think `spl_fstrans_unmark()` should be moved below after everything that can theoretically allocate memory.